pkg/storage-init/storage-init.sh

```diff
@@ -62,6 +62,14 @@ if P3=$(/hostfs/sbin/findfs PARTLABEL=P3) && [ -n "$P3" ]; then
         tune2fs -O encrypt "$P3"
         if ! mount -t ext4 -o dirsync,noatime "$P3" $PERSISTDIR; then
             echo "$(date -Ins -u) mount $P3 failed"
+        else
+            # The only bit of initialization we do is to point containerd to /persist
+            # The trick here is to only do it if /persist is available and otherwise
+            # allow containerd to run with /var/lib/containerd on tmpfs (to make sure
+            # that the system comes up somehow)
+            mkdir -p "$PERSISTDIR/containerd"
+            mkdir -p /hostfs/var/lib
+            ln -s "$PERSISTDIR/containerd" /hostfs/var/lib/containerd
         fi
     fi
 else
```

Comment:
Don't know what to do about the else case in L76 when no P3 partition exists. Clearly the device is badly installed, but driving on seems a poor choice. Adding the symlink on L76 doesn't seem useful but this PR raises the question about L76. Should be fine to make L76 fatal and exit

Reply:
Yeah, what is our policy if the device is badly installed? My preference is for killing it as well; I think the last thing we want is a device un a strange and unknown state.

Reply:
I've just discussed this part with @eriknordmark and it seems like we need to spend some time making storage-init more robust (and predictable) in general. Perhaps once @gkodali-zededa is done with logging bits he can take another look at storage-init. Still, I think this is orthogonal to the PR at hand -- so I'll just file a tracker story for @gkodali-zededa